opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/ServiceContextProvider.java

```diff
@@ -64,15 +74,41 @@
         if (null != authSession) {
             return authSession.getParent();
         } else {
-            ServiceSession session = new ServiceSession();
+            FacadeServiceableRequest facadeServiceable = request.getFacadeServiceable();
+            String sessionPassword = facadeServiceable.getSessionPassword();
+
+            if (facadeServiceable.getXRequestID() == null) {
+                throw new RuntimeException("no request id");
+            }
 
-            if (null != request.getFacadeServiceable().getServiceSessionId()) {
-                session.setId(request.getFacadeServiceable().getServiceSessionId());
+            Optional<ServiceSession> savedSession = serviceSessions.findById(facadeServiceable.getXRequestID());
+            if (savedSession.isPresent()) {
```

Comment:
Refactor this into a separate method, use Strings.isNullOrEmpty it makes no sense to add commons just for one function

Reply:
done